modules/platform/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/routing/adapter/WorkflowAdapter.java

```diff
@@ -50,13 +50,18 @@
     @POST
     public Response doPost(WorkflowRequest routingRequest) {
         DocumentModel doc = getTarget().getAdapter(DocumentModel.class);
-        final String workflowInstanceId = Framework.getService(DocumentRoutingService.class)
-                                                   .createNewInstance(routingRequest.getWorkflowModelName(),
-                                                           List.of(doc.getId()), routingRequest.getVariables(),
-                                                           ctx.getCoreSession(), true);
-        DocumentModel result = getContext().getCoreSession().getDocument(new IdRef(workflowInstanceId));
-        DocumentRoute route = result.getAdapter(DocumentRoute.class);
-        return Response.ok(route).status(Status.CREATED).build();
+        DocumentRoutingService documentRoutingService = Framework.getService(DocumentRoutingService.class);
+        if (documentRoutingService.canCreateInstance(doc.getCoreSession(), List.of(doc.getId()),
```

Comment:
Same comment about local variables than in `WorkflowObject`.

Reply:
done